docs/src/networking/custom-networking.md

```diff
@@ -300,6 +300,45 @@ spec:
       loadBalancerType: "lb"
 ```
 
+## Example spec to use custom role
+
+CAPOCI can be used to create Subnet/NSG in the VCN for custom workloads such as private load balancers,
+dedicated subnet for DB connection etc. The roles for such custom subnest must be defined as `custom`.
+The following spec shows an example for this scenario.
+
+```yaml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: OCICluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  compartmentId: "${OCI_COMPARTMENT_ID}"
+  networkSpec:
+    vcn:
+      name: ${CLUSTER_NAME}
+      subnets:
+        - name: db
+          role: custom
+          type: public
+          cidr: "172.16.5.0/28"
+      networkSecurityGroup:
+        list:
+          - name: db
+            role: custom
```

Comment:
Since we validated the subnet role this makes sense. What happens if the role does exist? The cluster will fail, but should we attempt to do any pre-flight sort of checks in the webhooks? or just let the logs tell the user why it failed?

Reply:
Sorry Joe, did not understand this. The role is called "custom". So the list of roles for a subnet in case of managed cluster are "control-plane", "worker", "lb-service" and "custom". The name can be anything. There cannot be a case where role does not exist right?

Reply:
aah ok, I think I understand. Our webhooks already fail in case the role does not exist. Our unit test covers that, the PR has that.